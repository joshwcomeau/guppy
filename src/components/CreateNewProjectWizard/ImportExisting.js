// @flow
import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import IconBase from 'react-icons-kit';
import { folderPlus } from 'react-icons-kit/feather/folderPlus';

import { COLORS, RAW_COLORS } from '../../constants';

import ImportProjectButton from '../ImportProjectButton';

type Props = {
  isOnboarding: boolean,
};

export const ImportExisting = ({ isOnboarding }: Props) => {
  if (isOnboarding) {
    // When the user is onboarding, there's a much more prominent prompt to
    // import existing projects, so we don't need this extra snippet.
    return null;
  }

  return (
    <Wrapper>
      <ImportProjectButton>
        <IconWrapper>
          <IconBase icon={folderPlus} size={18} />
        </IconWrapper>
      </ImportProjectButton>
      <MainText>
        Already have a project you'd like to manage with Guppy?{' '}
        <ImportProjectButton color={COLORS.textOnBackground}>
          Import it instead
        </ImportProjectButton>.
      </MainText>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  font-size: 1rem;
  max-width: 350px;
`;
const IconWrapper = styled.div`
  width: 42px;
  height: 42px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 2px solid rgba(255, 255, 255, 0.75);
  border-radius: 50%;
  color: ${RAW_COLORS.transparentWhite[300]};
`;

const MainText = styled.div`
  flex: 1;
  text-align: left;
  margin-left: 10px;
  color: ${RAW_COLORS.transparentWhite[300]};
`;

const mapStateToProps = state => ({
  isOnboarding: state.onboardingStatus !== 'done',
});

export default connect(mapStateToProps)(ImportExisting);
