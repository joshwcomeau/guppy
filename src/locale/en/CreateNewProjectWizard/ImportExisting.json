{
  "createNewProjectWizard.importExisting.main.alreadyHaveOne": "Already have a project you'd like to manage with Guppy? ",
  "createNewProjectWizard.importExisting.main.building": "Import it instead"
}