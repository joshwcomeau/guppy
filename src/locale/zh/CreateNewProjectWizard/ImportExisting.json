{
  "createNewProjectWizard.importExisting.main.alreadyHaveOne": "已经有一个项目并且想使用Guppy来管理？ ",
  "createNewProjectWizard.importExisting.main.building": "导入项目"
}