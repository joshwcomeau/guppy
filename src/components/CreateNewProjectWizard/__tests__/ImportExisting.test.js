import React from 'react';
import { shallow } from 'enzyme';
import { ImportExisting } from '../ImportExisting';

describe('ImportExisting component', () => {
  let wrapper;

  it('should render (after onboarding)', () => {
    wrapper = shallow(<ImportExisting isOnboarding={false} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`shouldn't render (during onboarding)`, () => {
    wrapper = shallow(<ImportExisting isOnboarding={true} />);
    expect(wrapper.html()).toBeNull();
  });
});
