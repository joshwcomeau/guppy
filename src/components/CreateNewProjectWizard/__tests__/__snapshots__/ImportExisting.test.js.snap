// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportExisting component should render (after onboarding) 1`] = `
<styled.div>
  <Connect(ImportProjectButton)>
    <styled.div>
      <Icon
        fill="currentColor"
        icon={
          Object {
            "attribs": Object {
              "fill": "none",
              "stroke": "currentColor",
              "stroke-linecap": "round",
              "stroke-linejoin": "round",
              "stroke-width": "2",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "d": "M22 19a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h5l2 3h9a2 2 0 0 1 2 2z",
                },
                "children": Array [],
                "name": "path",
              },
              Object {
                "attribs": Object {
                  "x1": "12",
                  "x2": "12",
                  "y1": "11",
                  "y2": "17",
                },
                "children": Array [],
                "name": "line",
              },
              Object {
                "attribs": Object {
                  "x1": "9",
                  "x2": "15",
                  "y1": "14",
                  "y2": "14",
                },
                "children": Array [],
                "name": "line",
              },
            ],
            "viewBox": "0 0 24 24",
          }
        }
        size={18}
      />
    </styled.div>
  </Connect(ImportProjectButton)>
  <styled.div>
    Already have a project you'd like to manage with Guppy?
     
    <Connect(ImportProjectButton)
      color="#FFF"
    >
      Import it instead
    </Connect(ImportProjectButton)>
    .
  </styled.div>
</styled.div>
`;
